model/manifest.go

```diff
@@ -153,6 +153,9 @@ type Manifest struct {
 
 	// Plugins can store any kind of data in Props to allow other plugins to use it.
 	Props map[string]interface{} `json:"props,omitempty" yaml:"props,omitempty"`
+
+	// Plugins can require system configurations
```

Comment:
Let's adopt godoc conventions here (noting we're violating them above...), and note the dependency on the helpers.
```suggestion
	// RequiredConfig defines any required configuration fields for the plugin to function.
	//
	// To enforce this configuration, use the helpers API in OnActivate:
	//     func (p *Plugin) OnActivate() error {
	//         if err := p.Helpers.HasRequiredConfig(); err != nil {
	//             return err
	//         }
	//     }
```